src/optimise1.jl

```diff
@@ -141,17 +138,30 @@ function minimise_icp(f::Function, x::IntervalBox{N, T}; reltol=1e-3, abstol=1e-
     return lb..global_minimum, arg_minima
 end
 
-function minimise_icp_constrained(f::Function, x::IntervalBox{N, T}, constraints::Vector{Constraint{T}} = Vector{Constraint{T}}(); reltol=1e-3, abstol=1e-3) where {N, T<:Real}
+
+struct ConstraintCond{T}
```

Comment:
Why not just `Constraint`?

Reply:
Their might be more than one constraint condition thats why we need to built vector of such conditions.